src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java

```diff
@@ -739,6 +747,343 @@ public boolean apply(@Nullable FlowNode input) {
         });
     }
 
+
+    private List<WorkspaceAction> getWorkspaceActions(WorkflowRun workflowRun) throws java.io.IOException{
+        FlowGraphWalker walker = new FlowGraphWalker(workflowRun.getExecution());
+        List<WorkspaceAction> actions = new ArrayList<WorkspaceAction>();
+        for (FlowNode n : walker) {
+            WorkspaceAction a = n.getAction(WorkspaceAction.class);
+            if (a != null) {
+                actions.add(a);
+            }
+        }
+        return actions;
+    }
+
+    /**
+     * Needs at least Jenkins version 2.145 which supports {@link ExecutorStepExecution.PlaceholderTask#getAffinityKey()}
+     */
+    @Issue("JENKINS-36547")
+    @Test public void reuseNodeFromPreviousRun() {
+        story.addStep(new Statement() {
```

Comment:
BTW `.then` is more friendly to Java 8.

```java
story.then(r -> {
    DumbSlave slave = r.createOnlineSlave();
    // etc.
});

Reply:
Thanks for mentioning. Indeed this looks much better. I'd propose to change it together with any possible additional changes requested.